osdk/src/commands/run.rs

```diff
@@ -9,19 +9,80 @@ use super::{build::create_base_and_build, util::DEFAULT_TARGET_RELPATH};
 use crate::{
     bundle::Bundle,
     config_manager::{BuildConfig, RunConfig},
+    error::Errno,
+    error_msg,
     util::{get_cargo_metadata, get_current_crate_info, get_target_directory},
 };
 
 pub fn execute_run_command(config: &RunConfig) {
+    if config.gdb_server_args.gdb_server {
+        use std::env;
+        env::set_var(
+            "RUSTFLAGS",
+            env::var("RUSTFLAGS").unwrap_or_default() + " -g",
+        );
+    }
+
     let ws_target_directory = get_target_directory();
     let osdk_target_directory = ws_target_directory.join(DEFAULT_TARGET_RELPATH);
     let target_name = get_current_crate_info().name;
     let default_bundle_directory = osdk_target_directory.join(target_name);
     let existing_bundle = Bundle::load(&default_bundle_directory);
 
+    let config = RunConfig {
+        manifest: {
+            if config.gdb_server_args.gdb_server {
+                let qemu_gdb_args: Vec<_> = {
+                    let gdb_stub_path = config.gdb_server_args.gdb_stub.as_str();
+                    match gdb::get_stub_type(gdb_stub_path) {
+                        gdb::StubType::Socket => {
```

Comment:
Technically, the Tcp option also uses sockets. So `StubType::Socket` should be `StubType::Unix`, which is short for UNIX sockets.